components/camel-ibm/camel-ibm-secrets-manager/src/main/java/org/apache/camel/component/ibm/secrets/manager/vault/IBMEventStreamReloadTriggerTask.java

```diff
@@ -161,50 +160,58 @@ public void run() {
-                                                lastReloadTime = Instant.now();
-                                                reload.onReload(this);
+                for (ConsumerRecord<String, String> record : records) {
+                    JsonNode recordJson;
+                    String secretType;
+                    try {
+                        recordJson = mapper.readTree(record.value());
+                    } catch (JsonProcessingException e) {
+                        throw new RuntimeException(e);
+                    }
+                    JsonNode payload = recordJson.get("data");
+                    if (payload != null) {
+                        secretType = payload.get("event_type").asText();
+                        if (secretType != null) {
+                            if (secretType.equalsIgnoreCase(IBM_SECRETS_MANAGER_SECRET_ROTATED_EVENT)) {
+                                ArrayNode secretsRotated = (ArrayNode) payload.get("secrets");
+                                for (JsonNode secret : secretsRotated) {
+                                    if (secret != null) {
+                                        String name = secret.get("secret_name").asText();
+                                        if (matchSecret(name)) {
+                                            updates.put(name, Instant.parse(secret.get("event_time").asText()));
+                                            if (isReloadEnabled()) {
+                                                LOG.info("Update for IBM secret: {} detected, triggering CamelContext reload",
+                                                        name);
+                                                triggerReloading = true;
                                             }
-                                            triggerReloading = false;
+                                            if (triggerReloading) {
+                                                ContextReloadStrategy reload
+                                                        = camelContext.hasService(ContextReloadStrategy.class);
+                                                if (reload != null) {
+                                                    // trigger reload
+                                                    lastReloadTime = Instant.now();
+                                                    reload.onReload(this);
+                                                }
+                                                triggerReloading = false;
+                                            }
+                                            break;
                                         }
-                                        break;
                                     }
                                 }
                             }
                         }
                     }
-                }
 
+                }
+            } catch (org.apache.kafka.common.errors.InterruptException ie) {
```

Comment:
@oscerd this one seems to be the only reason why the thread is interrupted. Can you double check?